src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -27372,5 +27368,554 @@ public ValueTuple(T1 item1)
+                    "TupleElementFieldSymbol: Item2" }, printFields(tuple3.TupleUnderlyingType));
+            }
+
+            static IEnumerable<string> printFields(NamedTypeSymbol type)
+            {
+                var fields = type.GetMembers().OfType<FieldSymbol>();
+                return fields.Select(s => s.GetType().Name + ": " + s.Name);
+            }
+
+            static string print(Symbol s)
+            {
+                return s.GetType().Name + ": " + s.ToTestDisplayString();
+            }
+        }
+
+        [Fact]
+        public void RealFieldsAreNotWrapped_LongTuples()
+        {
+            string source = @"
+public class C
+{
+    public (int, int, int, int, int, int, int, int) M() => throw null;
+    public (int Item1, int Item2, int Item3, int Item4, int Item5, int Item6, int Item7, int Item8) M2() => throw null;
+    public (int a, int b, int c, int d, int e, int f, int g, int h) M3() => throw null;
+}
+";
+            var comp = CreateCompilation(source + trivialRemainingTuples);
+            CompileAndVerify(comp, symbolValidator: verifyModule, sourceSymbolValidator: verifyModule);
+
+            static void verifyModule(ModuleSymbol module)
+            {
+                var tuple1 = (NamedTypeSymbol)module.GlobalNamespace.GetMember<MethodSymbol>("C.M").ReturnType;
+                Assert.Equal("ConstructedNamedTypeSymbol: (System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32, System.Int32)", print(tuple1));
+                AssertEx.SetEqual(new[] {
+                    "SubstitutedFieldSymbol: Rest",
+                    "TupleElementFieldSymbol: Item1",
+                    "TupleElementFieldSymbol: Item2",
+                    "TupleElementFieldSymbol: Item3",
+                    "TupleElementFieldSymbol: Item4",
+                    "TupleElementFieldSymbol: Item5",
+                    "TupleElementFieldSymbol: Item6",
+                    "TupleElementFieldSymbol: Item7",
+                    "TupleVirtualElementFieldSymbol: Item8" }, printFields(tuple1));
+
+                var tuple2 = (NamedTypeSymbol)module.GlobalNamespace.GetMember<MethodSymbol>("C.M2").ReturnType;
+                Assert.Equal("ConstructedNamedTypeSymbol: (System.Int32 Item1, System.Int32 Item2, System.Int32 Item3, System.Int32 Item4, System.Int32 Item5, System.Int32 Item6, System.Int32 Item7, System.Int32 Item8)", print(tuple2));
+                AssertEx.SetEqual(new[] {
+                    "SubstitutedFieldSymbol: Rest",
+                    "TupleElementFieldSymbol: Item1",
+                    "TupleElementFieldSymbol: Item2",
+                    "TupleElementFieldSymbol: Item3",
+                    "TupleElementFieldSymbol: Item4",
+                    "TupleElementFieldSymbol: Item5",
+                    "TupleElementFieldSymbol: Item6",
+                    "TupleElementFieldSymbol: Item7",
+                    "TupleVirtualElementFieldSymbol: Item8"
```

Comment:
>Item8 [](start = 53, length = 5)

How IsDefinition and other interesting APIs work for this symbol? Is there a difference in behavior with the previous case? What is it?

Reply:
I'll add direct verification on `IsDefinition` here. This and other APIs are covered in `DefaultAndFriendlyElementNames_*`.

---
In reply to: [426889458](https://github.com/dotnet/roslyn/pull/44231#discussion_r426889458) [](ancestors = 426889458)